src/Core/src/Platform/Android/EditTextExtensions.cs

```diff
@@ -239,5 +239,21 @@ internal static void SetInputType(this AppCompatEditText editText, IEntry entry)
 				editText.InputType = nativeInputTypeToUpdate;
 			}
 		}
+
+		public static void UpdateHorizontalTextAlignment(this AppCompatEditText textField, IEditor editor)
+		{
+			if (textField.Context!.HasRtlSupport())
+			{
+				// We want to use TextAlignment where possible because it doesn't conflict with the
+				// overall gravity of the underlying control
+				textField.TextAlignment = editor.HorizontalTextAlignment.ToTextAlignment();
+			}
+			else
+			{
+				// But if RTL support is not available for some reason, we have to resort
+				// to gravity, because Android will simply ignore text alignment
+				textField.Gravity = editor.HorizontalTextAlignment.ToHorizontalGravityFlags();
+			}
+       }
```

Comment:
I would suggest not adding this overload, but changing the existing `UpdateHorizontalTextAlignment` further up to maybe take a `ITextAlignment` instead of `IEntry`. That has additional code that might need to also run.

Reply:
Okay, I've removed the new `UpdateHorizontalTextAlignment`